libsolidity/analysis/TypeChecker.cpp

```diff
@@ -3844,13 +3846,23 @@ void TypeChecker::endVisit(Literal const& _literal)
 				)
 					// TODO: Is this triggered when the argument is out of range? Test.
 					parameterTypeMessage = "The type of the literal cannot be converted to the parameters of the suffix function.";
+
+				if (
+					auto expParam = dynamic_cast<IntegerType const*>(suffixFunctionType->parameterTypes().at(1));
+					expParam && expParam->isSigned()
+				)
+					exponentTypeMessage = "The type of the exponent must be unsigned for decimal literal suffix functions.";
+
 			}
 			else if (!literalType->isImplicitlyConvertibleTo(*suffixFunctionType->parameterTypes().front()))
 				parameterTypeMessage = "The type of the literal cannot be converted to the parameter of the suffix function.";
 
 			if (parameterTypeMessage.has_value())
 				m_errorReporter.typeError(8838_error, _literal.location(), parameterTypeMessage.value());
 
+			if (exponentTypeMessage.has_value())
+				m_errorReporter.typeError(3123_error, _literal.location(), exponentTypeMessage.value());
```

Comment:
Yeah, I saw that and I thought this situation (same error in two different places) would be more visible avoiding the report immediately, albeit it may look a little bit weird. Do you think that makes sense or is it too much?

Reply:
I think it's clearer if you report it immediately. I really wish I didn't have to do that with `parameterTypeMessage` either - I'd prefer to just have multiple `typeError()` calls.

Reply:
Ok, understood,  I'm going to change it